#!/usr/bin/env python
PACKAGE = "cloud_factory"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("width", double_t, 0, "Adjust the width of the random cloud", 1.0, 0.1, 10.0)
gen.add("height", double_t, 0, "Adjust the height of the random cloud", 1.0, 0.1, 10.0)
gen.add("resolution", double_t, 0, "Adjust the resolution of the random cloud", 0.1, 0.01, 2.0)
gen.add("roughness", double_t, 0, "Adjust the random roughness of cloud mode 3", 0.1, 0.01, 2.0)
gen.add("n_points", int_t, 0, "Number of random points for mode 1 ", 100, 10, 100000)
gen.add("gauss_A", double_t, 0, "Adjust the amplitude of the gaussian plane", 1.0, -10.0, 10.0);
gen.add("gauss_center_x", double_t, 0, "Adjust the center of the gaussian plane", 0.5, 0.0, 10.0);
gen.add("gauss_center_y", double_t, 0, "Adjust the center of the gaussian plane", 0.5, 0.0, 10.0);
gen.add("gauss_sigma_x",  double_t, 0, "Adjust the stddeviation of the gaussian plane", 1.0, 0.0, 10.0);
gen.add("gauss_sigma_y",  double_t, 0, "Adjust the stddeviation of the gaussian plane", 1.0, 0.0, 10.0);
gen.add("gauss_power_x",  double_t, 0, "Adjust the exponent of the gaussian plane", 1.0, 0.1, 10.0);
gen.add("gauss_power_y",  double_t, 0, "Adjust the exponent of the gaussian plane", 1.0, 0.1, 10.0);
gen.add("mode", int_t, 0, "Modes available 0, 1, 2, 3 ", 0, 0, 3)
exit(gen.generate(PACKAGE, "cloud_factory", "RandomClouds"))